charts/k8s-shredder/templates/deployment.yaml

```diff
@@ -0,0 +1,103 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "k8s-shredder.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "k8s-shredder.labels" . | indent 4 }}
+{{- if .Values.podAnnotations }}
+  annotations:
+{{ include "k8s-shredder.annotations" . | indent 4 }}
+{{- end }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+{{- with .Values.deploymentStrategy }}
+  strategy:
+{{- toYaml . | nindent 4 }}
+{{- end }}
+  selector:
+    matchLabels:
+{{ include "k8s-shredder.matchLabels" . | indent 6 }}
+  template:
+    metadata:
+      labels:
+{{ include "k8s-shredder.labels" . | indent 8 }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+      {{ include "k8s-shredder.annotations" . | indent 8 }}
+      {{- end }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ template "k8s-shredder.serviceAccountName" . }}
+      securityContext:
+      {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      initContainers:
+      {{- with .Values.initContainers }}
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+              {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```

Comment:
Personally I prefer the pattern of `{{ .Values.image.registry }}:{{ .Values.image.tag | default .Chart.AppVersion }}` and not setting the tag value in the values.yaml file.